airbyte-bootloader/src/main/java/io/airbyte/bootloader/SecretMigrator.java

```diff
@@ -0,0 +1,197 @@
+              definitionIdToSourceSpecs.get(source.getSourceDefinitionId())),
+              () -> UUID.randomUUID());
+          source.setConfiguration(migratedConfig);
+          return source;
+        })
+        .toList();
+
+    for (final SourceConnection source : sourceConnections) {
+      configPersistence.writeConfig(ConfigSchema.SOURCE_CONNECTION, source.getSourceId().toString(), source);
+    }
+  }
+
+  /**
+   * This is migrating the secrets for the destination actors
+   */
+  @VisibleForTesting
+  void migrateDestinations(final List<DestinationConnection> destinations, final Map<UUID, JsonNode> definitionIdToDestinationSpecs)
+      throws JsonValidationException, IOException {
+    log.info("Migration Destinations");
+
+    final List<DestinationConnection> destinationConnections = destinations.stream().map(destination -> {
+      final JsonNode migratedConfig = migrateConfiguration(new ConnectorConfiguration(
+          destination.getWorkspaceId(),
+          destination.getConfiguration(),
+          definitionIdToDestinationSpecs.get(destination.getDestinationDefinitionId())),
+          () -> UUID.randomUUID());
+      destination.setConfiguration(migratedConfig);
+      return destination;
+    })
+        .toList();
+    for (final DestinationConnection destination : destinationConnections) {
+      configPersistence.writeConfig(ConfigSchema.DESTINATION_CONNECTION, destination.getDestinationId().toString(), destination);
+    }
+  }
+
+  /**
+   * This is a generic method to migrate an actor configuration It will extract the secret path form
+   * the provided spec and then replace them by coordinates in the actor configuration
+   */
+  @VisibleForTesting
+  JsonNode migrateConfiguration(final ConnectorConfiguration connectorConfiguration, final Supplier<UUID> uuidProvider) {
+    if (connectorConfiguration.getSpec() == null) {
+      throw new IllegalStateException("No connector definition to match the connector");
+    }
+
+    final AtomicReference<JsonNode> connectorConfigurationJson = new AtomicReference<>(connectorConfiguration.getConfiguration());
+    final List<String> uniqSecretPaths = getSecretPath(connectorConfiguration.getSpec())
+        .stream()
+        .flatMap(secretPath -> getAllExplodedPath(connectorConfigurationJson.get(), secretPath).stream())
+        .toList();
+
+    final UUID workspaceId = connectorConfiguration.getWorkspace();
+    uniqSecretPaths.forEach(secretPath -> {
+      final Optional<JsonNode> secretValue = getValueForPath(connectorConfigurationJson.get(), secretPath);
+      if (secretValue.isEmpty()) {
+        throw new IllegalStateException("Missing secret for the path: " + secretPath);
+      }
+
+      // Only migrate plain text.
+      if (secretValue.get().isTextual()) {
+        final JsonNode stringSecretValue = secretValue.get();
+
+        final SecretCoordinate coordinate = new SecretCoordinate("airbyte_workspace_" + workspaceId + "_secret_" + uuidProvider.get(), 1);
```

Comment:
I think there could still be value in updating the SecretsHelper to have a simpler method that takes in a workspaceId and uuid and returns the SecretCoordinate with the coordinate name constructed correctly.

The reason I am arguing for this is because I want to avoid having the `"airbyte_workspace_" + workspaceId + "_secret_" + uuidProvider.get()` logic repeated in multiple places, since that could mean that if we want to change it at some point and we forget to update both places, we could be in an inconsistent state.